### PR TITLE
Add ERC: Token Revenue Sharing

### PR DESCRIPTION
**Pull request by quyphandang**
When opening a pull request to submit a new EIP, please use the suggested template: https://github.com/ethereum/EIPs/blob/master/eip-template.md

We have a GitHub bot that automatically merges some PRs. It will merge yours immediately if certain criteria are met:

 - The PR edits only existing draft PRs.
 - The build passes.
 - Your GitHub username or email address is listed in the 'author' header of all affected PRs, inside <triangular brackets>.
 - If matching on email address, the email address is the one publicly listed on your GitHub profile.

### CODE REVIEW

File: ERCS/erc-7254.md
Comment (SamWilsn) [inline ERCS/erc-7254.md:1]: Please rename this file `erc-7254.md -> eip-7254.md`.

File: assets/erc-7254/ERC7254.sol
Comment (SamWilsn) [inline assets/erc-7254/ERC7254.sol:1]: Assets still live in `assets/eip-7254` (our build system is a little weird :sweat_smile:)

**Review by SamWilsn — CHANGES_REQUESTED** [review]

File: ERCS/erc-7254.md
Comment (quyphandang) [reply ERCS/erc-7254.md:1]: I attempted to change the name to EIP, but the attempt failed. Consequently, I opted to reuse the old name ERC-7254.md.

File: assets/erc-7254/ERC7254.sol
Comment (quyphandang) [reply assets/erc-7254/ERC7254.sol:1]: I fixed it

**Review by SamWilsn — APPROVED** [review]
Assuming I haven't messed up the assets stuff again, this should be good to merge as a draft.

---

You'll really want to expand your Specification section to give implementers more information to go off of, for example, you should specify when events need to be emitted.

---

You should also include/link your `IERC7254` interface in the Specification section. It'll really help readers understand what's going on.

File: ERCS/erc-7254.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+## Reference Implementation
+
+* [`ERC7254`](../assets/eip-7254/ERC7254.sol)
+* [`IERC7254`](../assets/eip-7254/IERC7254.sol)
+
+## Security Considerations
+
+<!-- TODO -->
+
+Needs discussion.
```
Comment (SamWilsn) [inline ERCS/erc-7254.md:137]: Ah, I didn't realize this was going into Review. You'll need to complete the Rationale and Security Considerations sections before you can merge this.

**Review by SamWilsn — CHANGES_REQUESTED** [review]

**Pull request closed without merging**